lib/generic_equality.sail

```diff
@@ -49,11 +49,11 @@ $define _GENERIC_EQUALITY
 
 $include <flow.sail>
 
-val eq_anything = pure {ocaml: "(fun (x, y) -> x = y)", lem: "eq", coq: "generic_eq", _: "eq_anything"} : forall ('a : Type). ('a, 'a) -> bool
+val eq_anything = pure {ocaml: "(fun (x, y) -> x = y)", lem: "eq", coq: "generic_eq", lean: "BEq.beq", _: "eq_anything"} : forall ('a : Type). ('a, 'a) -> bool
```

Comment:
I would avoid `BEq` and just use `Eq` here and trust that all types we care about have an instance for `DecidableEq` which induces a cast from `Prop` to `Bool`.